src/serac/numerics/functional/tensor.hpp

```diff
@@ -1170,58 +1172,147 @@ SERAC_HOST_DEVICE bool is_symmetric_and_positive_definite(tensor<double, 3, 3> A
 }
 
 /**
- * @brief Solves Ax = b for x using Gaussian elimination with partial pivoting
- * @param[in] A The coefficient matrix A
- * @param[in] b The righthand side vector b
- * @note @a A and @a b are by-value as they are mutated as part of the elimination
+ * @brief Compute LU factorization of a matrix with partial pivoting
+ * @param[in] A The matrix to factorize
+ * @return x tuple of [P, L, U], with
+ * L the lower triangular factor
+ * U the upper triangular factor
+ * P an array indicating the row permutations due to pivoting.
+ * Row i of A was exchanged with row P[i].
  */
 template <typename T, int n>
-SERAC_HOST_DEVICE constexpr tensor<T, n> linear_solve(tensor<T, n, n> A, const tensor<T, n> b)
+SERAC_HOST_DEVICE constexpr auto lu(const tensor<T, n, n>& A)
 {
+  // BT question for Sam Mish: Why not use the standard library abs?
```

Comment:
I'm not sure-- if it had `auto` type in its argument it would make more sense. It's always annoying to manage the different `abs` implementations from `std::`, C, CUDA, and now our AD versions. If all the tests compile/pass without defining this `abs` then we should take it out.

Reply:
Question retracted. I'm leaving the lambda. I didn't think of the insufficient generality of the return type of `std::abs`